internal/server/passwordreset_test.go

```diff
@@ -0,0 +1,83 @@
+package server
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"gotest.tools/v3/assert"
+
+	"github.com/infrahq/infra/api"
+	"github.com/infrahq/infra/internal/server/data"
+	"github.com/infrahq/infra/internal/server/email"
+	"github.com/infrahq/infra/internal/server/models"
+)
+
+func TestPasswordResetFlow(t *testing.T) {
+	s := setupServer(t)
+	routes := s.GenerateRoutes(prometheus.NewRegistry())
+
+	email.TestMode = true
+
+	user := &models.Identity{
+		Name: "skeletor@example.com",
+	}
+
+	err := data.CreateIdentity(s.db, user)
+	assert.NilError(t, err)
+
+	err = data.CreateCredential(s.db, &models.Credential{
+		IdentityID:   user.ID,
+		PasswordHash: []byte("foo"),
+	})
+	assert.NilError(t, err)
+
+	// request password reset
```

Comment:
There's a `runStep` helper in this package that might be useful for these multi-step tests. It uses `t.Run`, but fails the parent test if any of the subtests fail.  It can be nice for multi-steps tests because the extra step name will be in the failure output, which gives the developer a little more information about which step failed.